Tools/GNUMake/comps/gnu.mak

```diff
@@ -13,25 +13,25 @@ F90FLAGS =
 
 ########################################################################
 
-gcc_version       := $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;')
-gcc_major_version := $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;' | sed -e 's;\..*;;')
-gcc_minor_version := $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;' | sed -e 's;[^.]*\.;;' | sed -e 's;\..*;;')
+gcc_version       = $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;')
+gcc_major_version = $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;' | sed -e 's;\..*;;')
+gcc_minor_version = $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;' | sed -e 's;[^.]*\.;;' | sed -e 's;\..*;;')
 
-COMP_VERSION := $(gcc_version)
+COMP_VERSION = $(gcc_version)
 
 DEFINES += -DBL_GCC_VERSION='$(gcc_version)'
 DEFINES += -DBL_GCC_MAJOR_VERSION=$(gcc_major_version)
 DEFINES += -DBL_GCC_MINOR_VERSION=$(gcc_minor_version)
 
 ########################################################################
 
-gcc_major_le_4 := $(shell expr $(gcc_major_version) \<= 4)
-gcc_minor_lt_8 := $(shell expr $(gcc_minor_version) \< 8)
-ifeq ($(gcc_major_le_4),1)
-  ifeq ($(gcc_minor_lt_8),1)
-    $(error GCC >= 4.8 required! Your version is $(gcc_version))
-  endif
-endif
+#gcc_major_le_4 = $(shell expr $(gcc_major_version) \<= 4)
+#gcc_minor_lt_8 = $(shell expr $(gcc_minor_version) \< 8)
+#ifeq ($(gcc_major_le_4),1)
+#  ifeq ($(gcc_minor_lt_8),1)
+#    $(warning GCC >= 4.8 required! Your version is $(gcc_version))
```

Comment:
This error was added because I ran on a system relatively recently where the default compiler was gcc 4.3, and in this case the code will not compile due to some things that depend on relatively recent gcc versions, so I think the error is still appropriate.

Reply:
The problem with the old way was that `Tools/GNUMake/comps/gnu.mak` sets "default" compilers and then tests against them:
```
CXX = g++
CC  = gcc
FC  = gfortran
F90 = gfortran

...

gcc_version       = $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;')
gcc_major_version = $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;' | sed -e 's;\..*;;')
gcc_minor_version = $(shell $(CXX) -dumpfullversion -dumpversion | head -1 | sed -e 's;.*  *;;' | sed -e 's;[^.]*\.;;' | sed -e 's;\..*;;')

...

gcc_major_le_4 = $(shell expr $(gcc_major_version) \<= 4)
gcc_minor_lt_8 = $(shell expr $(gcc_minor_version) \< 8)
ifeq ($(gcc_major_le_4),1)
  ifeq ($(gcc_minor_lt_8),1)
    $(error GCC >= 4.8 required! Your version is $(gcc_version))
  endif
endif
```

_before_ loading Make.local. The version test unfortunately forces GNUMakes lazy evaluation to evaluate `CXX` before the user's own compilers (in Make.local) have been specified. This isn't too much of a problem, if the system default gcc is > 4.8. For a Mac running homebrew this breaks because homebrew installs gcc to gcc-version_number (e.g. gcc-7). So then even if the user specifies the "correct" gcc in the Make.local, the code above breaks.

As an alternative, I suggest replacing a warning instead of an error. Something like:
"Your default gcc version is $(gcc_version). This might break during build. We therefore highly recommend that you specify GCC >= 4.8 in your Make.local. See the docs for an example"